src/rust/storage/seg/src/hashtable/mod.rs

```diff
@@ -109,6 +109,131 @@ pub(crate) struct HashTable {
     next_to_chain: u64,
 }
 
+struct Iter<'a> {
+    hashtable: &'a HashTable,
+    bucket_id: usize,
+    item_slot: usize,
+    chain_len: usize,
+    chain_idx: usize,
+    finished: bool,
+}
+
+struct IterMut<'a> {
+    ptr: *mut HashBucket,
+    bucket_id: usize,
+    item_slot: usize,
+    chain_len: usize,
+    chain_idx: usize,
+    finished: bool,
+    _marker: std::marker::PhantomData<&'a mut u64>,
```

Comment:
maybe a comment for the intention behind the phantom data would be helpful for future readers